source/plugins/go/src/extension/extension.go

```diff
@@ -40,64 +39,73 @@ func (e *Extension) GetOutputStreamId(datatype string) string {
+func (e *Extension) GetOutputNamedPipe(datatype string) string {
+	extensionconfiglock.Lock()
+	defer extensionconfiglock.Unlock()
+	if len(e.datatypeNamedPipeMap) > 0 && e.datatypeNamedPipeMap[datatype] != "" {
+		message := fmt.Sprintf("Named Pipe: %s for the datatype: %s", e.datatypeNamedPipeMap[datatype], datatype)
+		logger.Printf(message)
+		return e.datatypeNamedPipeMap[datatype]
+	}
+	var err error
+	e.datatypeNamedPipeMap, err = getExtensionDataTypeMapping(true)
+	if err != nil {
+		message := fmt.Sprintf("Error getting datatype to named pipe mapping: %s", err.Error())
+		logger.Printf(message)
+	}
+	return e.datatypeNamedPipeMap[datatype]
+}
+
+func getExtensionDataTypeMapping(isNamedPipe bool) (map[string]string, error) {
 	guid := uuid.New()
-	datatypeOutputStreamMap := make(map[string]string)
+	datatypeMap := make(map[string]string)
 
 	taggedData := map[string]interface{}{"Request": "AgentTaggedData", "RequestId": guid.String(), "Tag": "ContainerInsights", "Version": "1"}
 	jsonBytes, err := json.Marshal(taggedData)
 	// TODO: this error is unhandled
 
-	var data []byte
-	enc := codec.NewEncoderBytes(&data, new(codec.MsgpackHandle))
-	if err := enc.Encode(string(jsonBytes)); err != nil {
-		return datatypeOutputStreamMap, err
-	}
-
-	fs := &FluentSocket{}
-	fs.sockAddress = "/var/run/mdsd-ci/default_fluent.socket"
-	if containerType != "" && strings.Compare(strings.ToLower(containerType), "prometheussidecar") == 0 {
-		fs.sockAddress = fmt.Sprintf("/var/run/mdsd-%s/default_fluent.socket", containerType)
-	}
-	responseBytes, err := FluentSocketWriter.writeAndRead(fs, data)
-	defer FluentSocketWriter.disconnect(fs)
-	logger.Printf("Info::mdsd::Making call to FluentSocket: %s to write and read the config data", fs.sockAddress)
+	response, err := getExtensionConfigResponse(jsonBytes)
 	if err != nil {
-		return datatypeOutputStreamMap, err
+		return datatypeMap, err
 	}
-	response := string(responseBytes) // TODO: why is this converted to a string then back into a []byte?
 
-	var responseObjet AgentTaggedDataResponse
-	err = json.Unmarshal([]byte(response), &responseObjet)
+	var responseObject AgentTaggedDataResponse
+	err = json.Unmarshal([]byte(response), &responseObject)
 	if err != nil {
-		logger.Printf("Error::mdsd::Failed to unmarshal config data. Error message: %s", string(err.Error()))
-		return datatypeOutputStreamMap, err
+		logger.Printf("Error::mdsd/Windows AMA::Failed to unmarshal config data. Error message: %s", string(err.Error()))
+		return datatypeMap, err
 	}
 
 	var extensionData TaggedData
-	json.Unmarshal([]byte(responseObjet.TaggedData), &extensionData)
+	json.Unmarshal([]byte(responseObject.TaggedData), &extensionData)
 
 	extensionConfigs := extensionData.ExtensionConfigs
-	logger.Printf("Info::mdsd::build the datatype and streamid map -- start")
+	outputStreamDefinitions := make(map[string]StreamDefinition)
+	if isNamedPipe == true {

```

Comment:
schema of the extension config should be same for Linux and windows except that windows have streamid with namedpipe, right? if yes, why cant we have same implementation of this function which would work for both windows & linux?